internal/pkix/pkix.go

```diff
@@ -39,6 +38,10 @@ func ParseDistinguishedName(name string) (map[string]string, error) {
 			return nil, fmt.Errorf("distinguished name (DN) %q has multi-valued RDN attributes, remove multi-valued RDN attributes as they are not supported", name)
 		}
 		for _, attribute := range rdn.Attributes {
+			// stateOrProvince name 'S' is an alias for 'ST'
+			if attribute.Type == "S" {
+				attribute.Type = "ST"
+			}
```

Comment:
We need to add validation for not supporting both ST and S in same trusted identity.

Reply:
>We need to add validation for not supporting both ST and S in same trusted identity.

line 45-50 has confirmed that there is no duplicated S or ST
```go
			if attrKeyValue[attribute.Type] == "" {
				attrKeyValue[attribute.Type] = attribute.Value
			} else {
				return nil, fmt.Errorf("distinguished name (DN) %q has duplicate RDN attribute for %q, DN can only have unique RDN attributes", name, attribute.Type)
			}
```


> Also, certificate shouldn't contain S; do we need check for that?

Actually, there is not a string value "S" or "ST" written in certificate, instead, they are defined by OID, for example,
Here is an ANS.1 encoding of an certificate subject:
```sh
        SEQUENCE
        {
            SET
            {
                SEQUENCE
                {
                    OBJECT IDENTIFIER=CountryName (2.5.4.6)
                    PRINTABLE STRING='US'
                }
            }
            SET
            {
                SEQUENCE
                {
                    OBJECT IDENTIFIER=StateOrProvinceName (2.5.4.8)
                    PRINTABLE STRING='Washington'
                }
            }
            SET
            {
                SEQUENCE
                {
                    OBJECT IDENTIFIER=LocalityName (2.5.4.7)
                    PRINTABLE STRING='Redmond'
                }
            }
            SET
            {
                SEQUENCE
                {
                    OBJECT IDENTIFIER=OrganizationName (2.5.4.10)
                    PRINTABLE STRING='Microsoft Corporation'
                }
            }
            SET
            {
                SEQUENCE
                {
                    OBJECT IDENTIFIER=CommonName (2.5.4.3)
                    PRINTABLE STRING='Microsoft Time-Stamp PCA 2010'
                }
            }
        }
```
> OBJECT IDENTIFIER=StateOrProvinceName (2.5.4.8)

This OID is rendered as a string by the application; for Windows, it will be "S"; for Golang, it will be "ST". As a result, Golang will never see a certificate with the "S" field. It is only possible if the user writes "S" in the trust identity.